phoenix-core/src/main/java/org/apache/phoenix/index/GlobalIndexChecker.java

```diff
@@ -326,10 +326,10 @@ private void repairIndexRows(byte[] indexRowKey, long ts, List<Cell> row) throws
                 buildIndexScan.setAttribute(BaseScannerRegionObserver.REBUILD_INDEXES, TRUE_BYTES);
                 buildIndexScan.setAttribute(BaseScannerRegionObserver.SKIP_REGION_BOUNDARY_CHECK, Bytes.toBytes(true));
                 // Scan only columns included in the index table plus the empty column
```

Comment:
Is this comment still applicable ?

Reply:
Yes, it is. The difference is that in getAllColumns, you get index cols. The getAllColumnsForDataTable, you get the mapping columns in data table. For example: in index there is Col1 but in hbase it is in a cell with others with column qualifier:1. But in data table it is in its own cell with col qualifier Col1.